R/build-home-citation.R

```diff
@@ -31,39 +31,55 @@ read_citation <- function(path = ".") {
 data_home_sidebar_citation <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
 
-  if (!has_citation(pkg$src_path)) {
-    return(character())
-  }
   sidebar_section(
     heading = "Citation",
-    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html")
+    bullets = a(sprintf(tr_("Citing %s"), pkg$package), "authors.html#citation")
   )
 }
 
 data_citations <- function(pkg = ".") {
   pkg <- as_pkgdown(pkg)
-  cit <- read_citation(pkg$src_path)
-
-  text_version <- format(cit, style = "textVersion")
-  cit <- list(
-    html = ifelse(
-      text_version == "",
-      format(cit, style = "html"),
-      paste0("<p>", escape_html(text_version), "</p>")
-    ),
-    bibtex = format(cit, style = "bibtex")
-  )
 
-  purrr::transpose(cit)
+  if (has_citation(pkg$src_path)) {
+
+    cit <- read_citation(pkg$src_path)
+
+    text_version <- format(cit, style = "textVersion")
+    if (identical(text_version, "")) {
+      cit <- list(
+        html = format(cit, style = "html"),
+        bibtex = format(cit, style = "bibtex")
+      )
+    } else {
+      cit <- list(
+        html = paste0("<p>", text_version, "</p>"),
+        bibtex = format(cit, style = "bibtex")
+      )
+    }
+
+    return(purrr::transpose(cit))
+  }
+
+  autocit <- utils::packageDescription(pkg$package)
```

Comment:
I'd pull this code out into a pair of functions like `citation_default()` or `citation_included()` (but with better names if you have any ideas)